public/tabs/chat/messages/message_action.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiSmallButtonIcon, EuiCopy, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { IOutput, Interaction } from '../../../../common/types/chat_saved_object_attributes';
+import { useFeedback } from '../../../hooks/use_feed_back';
+import { HttpSetup } from '../../../../../../src/core/public';
+import { DataSourceService } from '../../../services/data_source_service';
+import { UsageCollectionSetup } from '../../../../../../src/plugins/usage_collection/public';
+
+interface MessageActionsProps {
+  contentToCopy?: string;
+  showRegenerate?: boolean;
+  onRegenerate?: () => void;
+  interaction?: Interaction | null;
+  message?: IOutput | null;
+  showFeedback?: boolean;
+  showTraceIcon?: boolean;
+  isOnTrace?: boolean;
+  traceInteractionId?: string;
+  onViewTrace?: () => void;
+  shouldActionBarVisibleOnHover?: boolean;
+  isFullWidth?: boolean;
+  httpSetup?: HttpSetup;
+  dataSourceService?: DataSourceService;
+  usageCollection?: UsageCollectionSetup;
+  metricAppName?: string;
+  buttonOrder?: string[];
+}
+
+type ButtonKey = 'copy' | 'regenerate' | 'thumbUp' | 'thumbDown' | 'trace';
+
+export const MessageActions: React.FC<MessageActionsProps> = ({
```

Comment:
This component seems very general. Not only for chat but also insight, or maybe other cases. So I think it should be moved out of chat to a more proper directory.

Reply:
How about put it into a new "action_button" under the "component" directory? Any suggestion?